### PR TITLE
Prevent prefetched Cart payload causing wrong values with cached Mini-Cart block

### PR DESCRIPTION
**Pull request by Aljullu**
In https://github.com/woocommerce/woocommerce-blocks/pull/9493, we updated the Mini-Cart block to support caching plugins. However, there was still an issue when hovering the button: cached values were displayed for a small period of time. That was caused by the prefetching we did of the `/cart/` endpoint, this PR removes that and updates the Mini-Cart block button to always display the correct values.

#### User Facing Testing

**With your admin user:**

1. Install _[WP-Optimize - Clean, Compress, Cache](https://wordpress.org/plugins/wp-optimize/)_ or a similar caching plugin.
2. Go to WP-Optimize > Settings > Cache and enable page caching.

**In a private/incognito window without being logged in:**

3. In the frontend visit any page. This will cache the page without products in the cart.

**With your admin user:**

4. Add some products to your cart.
5. Visit the same page from step 3.
6. Notice the Mini Cart totals do include the products you added in step 4, even though it's serving the cached version of step 3.
7. Hover the Mini-Cart button and verify the totals are still correct.

Before | After
--- | ---
[before.webm](https://github.com/woocommerce/woocommerce-blocks/assets/3616980/a7bed8d6-13d4-4b8a-b775-eea8feab9d32) | [after.webm](https://github.com/woocommerce/woocommerce-blocks/assets/3616980/c3b75b3b-60fb-4683-be6a-3bc9d108dd42)

8. Add another product to your cart.
9. Verify Mini-Cart totals updated correctly.
10. Navigate to any other page.
11. Verify Mini-Cart values are always correct and at no moment they render incorrect data.

* [ ] Do not include in the Testing Notes

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Prevent prefetched Cart payload causing wrong values with cached Mini-Cart block

### CODE REVIEW

File: assets/js/blocks/mini-cart/frontend.ts [hunk 1]
```diff
@@ -150,12 +150,10 @@ window.addEventListener( 'load', () => {
 		};
 
 		const openDrawerWithRefresh = () => {
-			miniCartBlock.dataset.isDataOutdated = 'true';
```
Comment (Aljullu) [inline assets/js/blocks/mini-cart/frontend.ts:153]: This change is not directly related to the rest of the PR, but I found out that `isDataOutdated` was defined but never read, so it should be safe to remove it.

**Comment by roykho** [comment]
I am still seeing a flash of the old data in some cases.


https://github.com/woocommerce/woocommerce-blocks/assets/2132595/a298e767-6828-4d8e-a42f-dd1ad7fb0fc7

**Steps to replicate:**

* Go to private/incognito window (not logged in).
* Go to shop page and add some products to the cart.
* Navigate to a different page and you will see a flash of the old price.

**Comment by Aljullu** [comment]
Oooh, good catch, @roykho! That's because when adding/removing products we weren't updating the values in the local storage, so in the next page load, we were using outdated data. It should have been fixed in f906b813ce2292c00b6615e7f07f37c0896bcce3. Would you mind taking another look?

**Review by roykho — APPROVED** [review]
Thanks for the quick fix, it is working great now. Good job!
